### PR TITLE
feat: SfColorPicker on product page

### LINKED ISSUE #949: [FEATURE]Product page- add side button '+ Colors' on mobile
**Describe how this feature will help us improve**
Easy and quick way of choosing color of product

**Describe the solution you'd considered**
Add side button '+ Colors' on mobile product page as below: 
plusColors
After pressing the '+ Colors' button a list of available colors of the product appears as below:
plusColors1
After pressing color, the choice is remembered as long as user is on that product page. Then if user adds the product to the cart, the product is added with chosen color.

**Acceptance criteria**

- [ ]  the '+ Colors' button is added to the product page on mobile
- [ ] after pressing the '+ Color' a list of available colors appears
- [ ] chosen color is remembered as long as the user is on that product page
- [ ] pressing the x sign at the bottom of the list closes it
- [ ] after choosing the color, when user adds the product to the cart, it is added with tat chosen color

### PR DESCRIPTION
**Pull request by justyna-13**
# Related issue
Closes #949 

# Scope of work
SfColorPicker added on product page 

# Screenshots of visual changes
Zrzut ekranu z 2020-11-06 08-39-56
Zrzut ekranu z 2020-11-06 08-40-12
Zrzut ekranu z 2020-11-06 08-40-24

# Checklist

- [ ] No commented blocks of code left
- [ ] Run tests and docs
- [ ] Self code-reviewed
- [ ] Changes documented 

If applicable:

- [ ] I followed [composition rules](https://docs.storefrontui.io/contributing/coding-guidelines.html#component-rules) for my component
- [ ] I tested the component in most common device sizes (can be tested in Storybook [from viewport addon in top menu](https://github.com/storybooks/storybook/tree/master/addons/viewport))

### CODE REVIEW

**Review by AdamPawlinski — COMMENTED** [review]
However there is no such design on figma, but I wonder if the color picker should be aligned to the right edge of gallery or picture when you are on mobile mode. Or maybe gallery should take 100% width: 
image

What do you think? @justyna-13

File: packages/vue/src/components/organisms/SfProductCard/SfProductCard.stories.js [hunk 1]
```diff
@@ -83,13 +83,44 @@ storiesOf("Organisms|ProductCard", module)
       alert(label) {
         alert(label);
       },
+      handleSelectedColor(color) {
```
Comment (Baroshem) [inline packages/vue/src/components/organisms/SfProductCard/SfProductCard.stories.js:86]: Two things here:

1. Maybe small naming convention change would make this method easier to understand (`color` => `selectedColor`, `el` => `color`
2. Currently we do not have many colors however in the future there may be more. Here we are looking for one specific color and we are toggling its value and for anything else we are setting `false`. This is not very efficient. The better approach in my opinion would be to find one (first) element of the colors array and toggle it's value.

```js
      handleSelectedColor(selectedColor) {
        const pickedColorIndex = this.colors.findIndex(color => color.label === selectedColor.label);  // here we are returning first element index that matches the query

        this.colors[pickedColorIndex].selected = !this.colors[pickedColorIndex].selected
      },
```

What do you think?

File: packages/vue/src/components/organisms/SfProductCard/SfProductCard.stories.js [hunk 2]
```diff
@@ -181,13 +213,42 @@ storiesOf("Organisms|ProductCard", module)
...
```
Comment (Baroshem) [inline packages/vue/src/components/organisms/SfProductCard/SfProductCard.stories.js:216]: Here the same as in the previous comment

File: packages/vue/src/components/organisms/SfProductCard/SfProductCard.stories.js [hunk 3]
```diff
@@ -279,13 +341,42 @@ storiesOf("Organisms|ProductCard", module)
...
```
Comment (Baroshem) [inline packages/vue/src/components/organisms/SfProductCard/SfProductCard.stories.js:344]: Same here

File: packages/vue/src/components/organisms/SfProductCard/SfProductCard.stories.js [hunk 4]
```diff
@@ -385,19 +477,44 @@ storiesOf("Organisms|ProductCard", module)
...
+            selected: false,
+          },
+        ],
       };
     },
     methods: {
       alert(label) {
         alert(label);
       },
+      handleSelectedColor(color) {
```
Comment (Baroshem) [inline packages/vue/src/components/organisms/SfProductCard/SfProductCard.stories.js:503]: refactor

File: packages/vue/src/components/organisms/SfProductCard/SfProductCard.vue [hunk 5]
```diff
@@ -27,6 +27,24 @@
...
+          :is-open="!isMobile || openColorPicker"
+          @click:toggle="toggleColorPicker"
+        >
+          <SfColor
+            v-for="(color, i) in colors"
+            :key="color.value"
+            :color="color.color"
+            :selected="color.selected"
+            class="sf-product-card__color"
+            @click="handleSelectedColor(i)"
```
Comment (Baroshem) [inline packages/vue/src/components/organisms/SfProductCard/SfProductCard.vue:44]: Can't we just emit the event with color here?
```suggestion
            @click="$emit("click:colors", color)"
```

File: packages/vue/src/examples/pages/product/Product.vue [hunk 6]
```diff
@@ -6,6 +6,21 @@
...
+        :has-close="true"
+        @click:toggle="toggleColorPicker"
+      >
+        <SfColor
+          v-for="(color, i) in product.colors"
+          :key="color.name"
+          class="product__color"
+          :color="color.color"
+          :selected="color.selected"
+          @click="selectColor(i)"
```
Comment (Baroshem) [inline packages/vue/src/examples/pages/product/Product.vue:21]: I cannot find this method. Was it developed earlier?

**Review by Baroshem — CHANGES_REQUESTED** [review]
Overall it looks good, some small adjustments from my side ;) Also, please resolve conflicts.

**Pull request closed without merging**